src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutes.scala

```diff
@@ -199,6 +199,23 @@ trait ResourceRoutes extends SamUserDirectives with SecurityDirectives with SamM
                     }
                   }
                 } ~
+                pathPrefix("memberPolicies") {
+                  requireActionsForSharePolicy(policyId, samUser, samRequestContext) {
+                    pathPrefix(Segment) { memberResourceType =>
+                      pathPrefix(Segment) { memberResourceId =>
+                        val memberResource = FullyQualifiedResourceId(ResourceTypeName(memberResourceType),
+                          ResourceId(memberResourceId))
+                        pathPrefix(Segment) { memberPolicyName =>
+                          val subject = FullyQualifiedPolicyId(memberResource, AccessPolicyName(memberPolicyName))
+                          pathEndOrSingleSlash {
+                            putUserInPolicy(policyId, subject, samRequestContext) ~
+                            deleteUserFromPolicy(policyId, subject, samRequestContext)
+                          }
+                        }
+                      }
+                    }
```

Comment:
for code cleanliness, how about consolidating the three nested `pathPrefix(Segment)` and the final `pathEndOrSingleSlash` into one `path(…)` directive?

```suggestion
                  path(Segment / Segment / Segment) { (memberResourceType, memberResourceId, memberPolicyName) =>
                    val memberResource = FullyQualifiedResourceId(ResourceTypeName(memberResourceType),
                      ResourceId(memberResourceId))
                    val subject = FullyQualifiedPolicyId(memberResource, AccessPolicyName(memberPolicyName))
                    putUserInPolicy(policyId, subject, samRequestContext) ~
                      deleteUserFromPolicy(policyId, subject, samRequestContext)
                  }
```

note that this is slightly more strict - it does NOT match a slash at the end of the API, whereas the existing PR code using `pathEndOrSingleSlash` does allow a slash at the end. In my opinion this is ok; we don't need to allow a trailing slash.

Reply:
pushing back on the `pathEndOrSingleSlash` for consistency and slight usability improvement (I know I'm someone who would have/not have a slash at the end of a path in my config, and watch a deployment break over it)